internal/pkg/agent/cmd/uninstall.go

```diff
@@ -86,28 +84,13 @@ func uninstallCmd(streams *cli.IOStreams, cmd *cobra.Command) error {
 
 	progBar := install.CreateAndStartNewSpinner(streams.Out, "Uninstalling Elastic Agent...")
 
-	logCfg := logp.DefaultConfig(logp.DefaultEnvironment)
-	logCfg.Level = logp.DebugLevel
-	// Using in memory logger, so we don't write logs to the
-	// directory we are trying to delete
-	logp.ToObserverOutput()(&logCfg)
-
-	err = logp.Configure(logCfg)
-	if err != nil {
-		return fmt.Errorf("error creating logging config: %w", err)
-	}
-
-	log := logger.NewWithoutConfig("")
-
+	log, logBuff := logger.NewInMemory("uninstall")
 	defer func() {
 		if err == nil {
 			return
 		}
-		oLogs := logp.ObserverLogs().TakeAll()
 		fmt.Fprintf(os.Stderr, "Error uninstalling. Printing logs\n")
-		for _, oLog := range oLogs {
-			fmt.Fprintf(os.Stderr, "%v\n", oLog.Entry)
-		}
+		fmt.Fprintf(os.Stderr, logBuff.String())
```

Comment:
As per the static check this should be (same for install.go):
```suggestion
		fmt.Fprint(os.Stderr, logBuff.String())
```